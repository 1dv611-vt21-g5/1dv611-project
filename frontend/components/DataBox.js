import { Box, Flex, Text } from '@chakra-ui/react'

const DataBox = ({ setShowData, data }) => {
  return (
    <Box cursor="pointer" onClick={() => setShowData(false)}>
      {Object.entries(data).map(([key, value]) => (
        <Flex>
          <Text fontSize="xs" mr="0.5rem" fontWeight="bold">{key}:</Text>
          <Text fontSize="xs" textAlign="left" maxW="40rem" isTruncated>{JSON.stringify(value)}</Text>
        </Flex>
      ))}
    </Box>
  )
}

export default DataBox