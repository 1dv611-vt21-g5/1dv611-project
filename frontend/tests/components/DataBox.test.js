import React from 'react'
import { render } from '@testing-library/react'
import DataBox from '../../components/DataBox'

it('Render snapshot correctly', () => {
  const data = { person: 'johannes', age: 28 }
  const { container } = render(<DataBox data={data} />)
  expect(container).toMatchSnapshot()
})

it('Renders a different snapshot correctly', () => {
  const data = { person: 'Adam', age: 31 }
  const { container } = render(<DataBox data={data} />)

  expect(container).toMatchSnapshot()
})
