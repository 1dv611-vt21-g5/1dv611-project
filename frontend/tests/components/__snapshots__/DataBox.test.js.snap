// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render snapshot correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-fade"
      style="opacity: 0;"
    >
      <div
        class="css-1vxu83e"
      >
        <svg
          baseProfile="tiny"
          class="chakra-icon css-11xo6sj"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M17 21h-10c-1.654 0-3-1.346-3-3v-12c0-1.654 1.346-3 3-3h10c1.654 0 3 1.346 3 3v12c0 1.654-1.346 3-3 3zm-10-16c-.551 0-1 .449-1 1v12c0 .551.449 1 1 1h10c.551 0 1-.449 1-1v-12c0-.551-.449-1-1-1h-10zM16 11h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 8h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 14h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 17h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5z"
            />
          </g>
        </svg>
        <p
          class="chakra-text css-1lj023j"
        >
          Click to display the raw device info sent by this device.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders a different snapshot correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-fade"
      style="opacity: 0;"
    >
      <div
        class="css-1vxu83e"
      >
        <svg
          baseProfile="tiny"
          class="chakra-icon css-11xo6sj"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M17 21h-10c-1.654 0-3-1.346-3-3v-12c0-1.654 1.346-3 3-3h10c1.654 0 3 1.346 3 3v12c0 1.654-1.346 3-3 3zm-10-16c-.551 0-1 .449-1 1v12c0 .551.449 1 1 1h10c.551 0 1-.449 1-1v-12c0-.551-.449-1-1-1h-10zM16 11h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 8h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 14h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5zM16 17h-8c-.276 0-.5-.224-.5-.5s.224-.5.5-.5h8c.276 0 .5.224.5.5s-.224.5-.5.5z"
            />
          </g>
        </svg>
        <p
          class="chakra-text css-1lj023j"
        >
          Click to display the raw device info sent by this device.
        </p>
      </div>
    </div>
  </div>
</div>
`;
